### PR TITLE
Adapt to removed support for loading pack.gz packaged artifacts in P2

### PR DESCRIPTION
**Pull request by HannesWell**
Eclipse P2 got its support for loading pack.gz packed artifacts from repositories finally removed in
https://github.com/eclipse-equinox/p2/pull/310.

In order to fix the tests, remove packed duplicates from and replace only packed ecf artifact with unpacked eclipse.core.runtime artifact in test repository at `\resources\repositories\packgz`.

This is required for https://github.com/eclipse-tycho/tycho/pull/3043 respectively https://github.com/eclipse-tycho/tycho/pull/3125.

### CODE REVIEW

**Comment by HannesWell** [comment]
The LocalArtifactRepositoryP2APITest still needs some adjustments since it explicitly relays on packed artifacts.
Will continue tomorrow.

**Comment by laeubi** [comment]
Maybe we should keep the pack.gz items in the repository to tests they are really ignored and just adjust the number in the assertions? If a test relies on packgz to work and can't be adjusted it is maybe time to delete it.

**Comment by akurtakov** [comment]
@HannesWell Are you gonna be working on this one today?

**Comment by HannesWell** [comment]
I will check which tests not make sense anymore and remove or adjust them based on the gone support of packgz.

> @HannesWell Are you gonna be working on this one today?

I'm sorry but I won't have time for this this evening, but will continue tomorrow.

**Comment by HannesWell** [comment]
I now also adapted the test cases in LocalArtifactRepositoryP2APITest to reflect the fact that there a no two descriptors/real artifacts for the same artifact-key anymore (I'm not aware of a scenario where this can happen any more now that pack.gz is gone).
Test cases that are pointless with only one artifact per key are now removed. 
Locally all cases in that class passed, lets hope that they also pass in the CI and that there are no more cases that fail.

**Comment by laeubi** [comment]
> I'm not aware of a scenario where this can happen any more now that pack.gz is gone).

It is not really gone... all already released updatesites still contain such artifacts thats why they should be ignored, that means even if a repository contains pack.gz it is simply not used.

**Comment by HannesWell** [comment]
I'm puzzled why FeatureRootfileArtifactRepositoryTest failed in the GH Linux workflow?
Locally it fails for me to, but even on the current master. So I think this is not related to my change.
Will re-start the workflow once all the others terminated.

> > I'm not aware of a scenario where this can happen any more now that pack.gz is gone).
> 
> It is not really gone... all already released updatesites still contain such artifacts thats why they should be ignored, that means even if a repository contains pack.gz it is simply not used.

Of course. I expressed myself unclear. Let me rephrase it: Now that pack.gz artifacts are not read/considered anymore.

**Comment by laeubi** [comment]
Yes Github Linux fails but unless Jenkins linux succeeds its not a problem I want to take a look once we have merged all the cleanup/upgrade stuff, it seems the test must somehow be a TychoPlexusTestcase now because "something" triggers access to Extension registry.

**Comment by HannesWell** [comment]
> Yes Github Linux fails but unless Jenkins linux succeeds its not a problem I want to take a look once we have merged all the cleanup/upgrade stuff, it seems the test must somehow be a TychoPlexusTestcase now because "something" triggers access to Extension registry.

OK, good to know.
Then I will submit this if all other checks (and https://github.com/eclipse-tycho/tycho/pull/3043, which also contains this) pass so that we can continue with the back-port.

**Comment by HannesWell** [comment]
Jenkins almost fully succeeded, only the following integration tests failed

- org.eclipse.tycho.test.p2Repository.P2RepositoryDownloadTest.testReactorCanBeVerified
- org.eclipse.tycho.test.reactor.makeBehaviour.MavenReactorMakeOptionsTest.testCompleteBuild

But they also fail on the master branch so its not related.
Therefore submitting this now.
